priv/apps/itest/lib/in_flight_exit_client.ex

```diff
@@ -0,0 +1,262 @@
+defmodule Itest.InFlightExitClient do
+  @moduledoc """
+  Client to run the in flight exit flow
+  """
+
+  alias Itest.Client
+  alias Itest.Transactions.Currency
+  alias Itest.Transactions.Encoding
+  alias Itest.Transactions.PaymentType
+  alias WatcherSecurityCriticalAPI.Api.InFlightExit
+  alias WatcherSecurityCriticalAPI.Connection, as: Watcher
+  alias WatcherSecurityCriticalAPI.Model.InFlightExitTxBytesBodySchema
+
+  import Itest.Poller, only: [wait_on_receipt_confirmed: 2]
+  require Logger
+
+  @ife_bond_size 37_000_000_000_000_000
+  @ife_gas 2_000_000
+  @ife_gas_price 1_000_000_000
+  @retry_count 120
+  @min_exit_period 50_000
+  @gas_process_exit 5_712_388
+  @gas_process_exit_price 1_000_000_000
+  @gas_add_exit_queue 800_000
+
+  @gas_piggyback 1_000_000
+  @piggyback_bond 28_000_000_000_000_000
+
+  defstruct [
+    :add_exit_queue_hash,
+    :address,
+    :address_key,
+    :exit_data,
+    :exit_game_contract_address,
+    :exit_id,
+    :piggyback_input_hash,
+    :process_exit_receipt_hash,
+    :receiver_address,
+    :sign_hash,
+    :signed_txbytes,
+    :start_in_flight_exit,
+    :txbytes
+  ]
+
+  def start_in_flight_exit(sender_address, sender_key, receiver_address) do
+    %__MODULE__{address: sender_address, address_key: sender_key, receiver_address: receiver_address}
+    |> create_transaction()
+    |> sign_transaction()
+    |> add_exit_queue()
+    |> get_exit_data()
+    |> get_exit_game_contract_address()
+    |> do_in_flight_exit()
+    |> get_in_flight_exit_id()
+    |> piggyback_input()
+    |> process_exit()
+  end
+
+  defp get_exit_data(%{signed_txbytes: txbytes} = se) do
+    payload = %InFlightExitTxBytesBodySchema{txbytes: txbytes}
+    {:ok, response} = InFlightExit.in_flight_exit_get_data(Watcher.new(), payload)
+
+    exit_data = Jason.decode!(response.body)["data"]
+    %{se | exit_data: exit_data}
+  end
+
+  defp create_transaction(%{address: address, receiver_address: receiver_address} = se) do
+    {:ok, [sign_hash, _typed_data, txbytes]} =
+      Client.create_transaction(Currency.to_wei(1), address, receiver_address, Currency.ether())
+
+    %{se | txbytes: txbytes, sign_hash: sign_hash}
+  end
+
+  defp sign_transaction(%{address_key: address_key, txbytes: txbytes, sign_hash: sign_hash} = se) do
+    [type, inputs, outputs, metadata] = txbytes |> Encoding.to_binary() |> ExRLP.decode()
```

Comment:
wouldn't it be better if we introduce a common struct for this?

Reply:
@InoMurko sure, if we want to go with that approach though I'd suggest we just use `ex_plasma` at this point as it handles all of this.

Reply:
I'm afraid to include ExPlasma if thats the only thing we're able to use it for... Is that true?